x/gamm/keeper/pool.go

```diff
@@ -79,6 +79,80 @@ func (k Keeper) SetPool(ctx sdk.Context, pool types.PoolI) error {
 	return nil
 }
 
+// CleanupBalancerPool destructs a pool and refund all the assets according to
+// the shares held by the accounts. CleanupBalancerPool should be called not during
+// the chain execution time, as it iterates the entire account balances.
+//
+// All locks on this pool share must be unlocked in prior. Execute LockupKeeper.ForceUnlock
+// on remaning locks before calling this function.
+func (k Keeper) CleanupBalancerPool(ctx sdk.Context, poolId uint64) (err error) {
+	pool, err := k.GetPool(ctx, poolId)
+	if err != nil {
+		return err
+	}
+
+	totalShares := pool.GetTotalShares().Amount
+	shareDenom := pool.GetTotalShares().Denom
+	poolAssets := pool.GetAllPoolAssets()
+
+	// first iterate through the share holders and burn them
+	k.bankKeeper.IterateAllBalances(ctx, func(addr sdk.AccAddress, coin sdk.Coin) (stop bool) {
+		if coin.Denom != shareDenom || coin.Amount.IsZero() {
+			return
+		}
+
+		shareAmount := coin.Amount
```

Comment:
lets alias coin to `userPoolShare` here. I think it makes a lot of subsequent lines easier to read